### DISCUSSION TITLE
predict without random effect(s)?

### DISCUSSION THREAD

**Discussion by jbigman-NOAA** [vast-lib/tinyVAST#52] [Q&A]
Is there a way to exclude the spatial/spatiotemporal random effects when predicting (e.g., predict(object, re_form = NA)? I'm plotting predictions and they change quite a bit depending on what grid cell I use when predicting. Thanks!

**Comment by James-Thorson-NOAA** [comment]
Do you want SEs too or just raw values?

On Friday, September 20, 2024, jbigman-NOAA ***@***.***>
wrote:

> Is there a way to exclude the spatial/spatiotemporal random effects when
> predicting (e.g., predict(object, re_form = NA)? I'm plotting predictions
> and they change quite a bit depending on what grid cell I use when
> predicting. Thanks!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vast-lib/tinyVAST/discussions/52>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL62VMUNGS6KMRRNMFJJ73LZXR74TAVCNFSM6AAAAABOSZPXSSVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGIYTGMBVGQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
James Thorson (he/him)

Alaska Fisheries Science Center, NMFS
Affiliate Faculty, University of Washington

**Reply by jbigman-NOAA** [reply]
Raw values just for now would be great, thanks!

**Comment by James-Thorson-NOAA** [comment]
Using `tinyVAST@dev` branch I just updated the reference docs, please see `?predict`. so e.g. `predict(fit,what="palpha_g")` is the component of the 1st linear predictor from fixed covariates.  

You just want fixed covariates, or also spline effects, or ...?

**Comment by jbigman-NOAA** [comment]
I think I just need the fixed covariates. Do I need to rerun models to get this to work? I have no issues when re-fitting a model then calling ```predict(fit, what = "palpha_g")``` but with saved models I get this error:

```Error in getParameterOrder(data, parameters, new.env(), DLL = DLL) : 
  Error when reading the variable: 'weights_i'. Please check data and parameters.
In addition: Warning message:
In getParameterOrder(data, parameters, new.env(), DLL = DLL) :
  Expected object. Got NULL.
```

**Comment by James-Thorson-NOAA** [comment]
There's other changes on the dev branch like a new weights argument. You
could revert if you saved you're previous commit record or ZIP, or rerun
with the dev, sorry!

On Friday, September 20, 2024, jbigman-NOAA ***@***.***>
wrote:

> I think I just need the fixed covariates. Do I need to rerun models to get
> this to work? I have no issues when re-fitting a model then calling predict(fit,
> what = "palpha_g") but with saved models I get this error:
>
>   Error when reading the variable: 'weights_i'. Please check data and parameters.
> In addition: Warning message:
> In getParameterOrder(data, parameters, new.env(), DLL = DLL) :
>   Expected object. Got NULL.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vast-lib/tinyVAST/discussions/52#discussioncomment-10709769>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL62VMV2AAXAFMCCH6WVVYTZXSKSZAVCNFSM6AAAAABOSZPXSSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANZQHE3TMOI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
James Thorson (he/him)

Alaska Fisheries Science Center, NMFS
Affiliate Faculty, University of Washington

**Comment by jbigman-NOAA** [comment]
No problem! I can reinstall the main branch and sometime closer to submission rerun the models with the dev branch and then predict and plot again. Thanks for adding this so quickly!

**Comment by James-Thorson-NOAA** [comment]
It's in the main branch too, just not in the docs there

On Friday, September 20, 2024, jbigman-NOAA ***@***.***>
wrote:

> No problem! I can reinstall the main branch and sometime closer to
> submission rerun the models with the dev branch and then predict and plot
> again. Thanks for adding this so quickly!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vast-lib/tinyVAST/discussions/52#discussioncomment-10709859>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL62VMTFJDTMJN3ITBAA74TZXSNCPAVCNFSM6AAAAABOSZPXSSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANZQHE4DKOI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
James Thorson (he/him)

Alaska Fisheries Science Center, NMFS
Affiliate Faculty, University of Washington

**Comment by jbigman-NOAA** [comment]
Ok, no worries. I'll rerun at least the most supported models so I can plot them. Thanks!
